### PR TITLE
Disallow table redirections for GRANT, REVOKE, SET TABLE AUTHORIZATION, DENY table tasks

### PR DESCRIPTION
**Pull request by findinpath**
## Description

As discussed on https://github.com/trinodb/trino/issues/11202 the statements which deal with table permission statements:

- `GRANT`
- `REVOKE`
- `ALTER TABLE ... SET AUTHORIZATION`

> Is this change a fix, improvement, new feature, refactoring, or other?

This PR adds redirection awareness for the table permission statements.

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

This is a change to the core query engine.

> How would you describe this change to a non-technical end user or system administrator?

Disallow table redirections for GRANT, REVOKE, SET TABLE AUTHORIZATION table tasks.

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
(x) Release notes entries required with the following suggested text:

```markdown
# Main
* Disallow table redirections for GRANT, REVOKE, SET TABLE AUTHORIZATION table tasks
```

### CODE REVIEW

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveRedirectionToIceberg.java [hunk 1]
```diff
@@ -461,6 +461,21 @@ public void testSystemJdbcColumns()
         onTrino().executeQuery("DROP SCHEMA hive." + schemaName);
     }
 
+    @Test(groups = {HIVE_ICEBERG_REDIRECTIONS, PROFILE_SPECIFIC_TESTS})
```
Comment (kokosing) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveRedirectionToIceberg.java:464]: We should have also a test in `io.trino.security.TestAccessControl`

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveRedirectionToIceberg.java [hunk 2]
```diff
@@ -476,6 +476,21 @@ public void testGrant()
         onTrino().executeQuery("DROP TABLE " + icebergTableName);
     }
 
+    @Test(groups = {HIVE_ICEBERG_REDIRECTIONS, PROFILE_SPECIFIC_TESTS})
+    public void testRevoke()
```
Comment (kokosing) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveRedirectionToIceberg.java:480]: We should have also a test in `io.trino.security.TestAccessControl`

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveRedirectionToIceberg.java [hunk 3]
```diff
@@ -491,6 +491,21 @@ public void testRevoke()
         onTrino().executeQuery("DROP TABLE " + icebergTableName);
     }
 
+    @Test(groups = {HIVE_ICEBERG_REDIRECTIONS, PROFILE_SPECIFIC_TESTS})
+    public void testSetTableAuthorization()
```
Comment (kokosing) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveRedirectionToIceberg.java:495]: We should have also a test in `io.trino.security.TestAccessControl`

**Review by kokosing — APPROVED** [review]

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveRedirectionToIceberg.java (hunk 2 above)
Comment (findinpath) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveRedirectionToIceberg.java:480]: The `ServerMainModule` uses by default `DisabledSystemSecurityMetadata` which doesn't have support for granting, revoking, denying permissions.

The exception message logged from the test:

```
Catalog does not support permission management: blackhole
```

**Review by kokosing — APPROVED** [review]

**Pull request merged**